addons/website_sale/models/sale_order.py

```diff
@@ -153,7 +153,10 @@ def _cart_update(self, product_id=None, line_id=None, add_qty=0, set_qty=0, **kw
         SaleOrderLineSudo = self.env['sale.order.line'].sudo().with_context(product_context)
         # change lang to get correct name of attributes/values
         product_with_context = self.env['product.product'].with_context(product_context)
-        product = product_with_context.browse(int(product_id))
+        product = product_with_context.browse(int(product_id)).exists()
+
+        if not product or not product._is_add_to_cart_allowed():
```

Comment:
why you have not use ?

`product = product_with_context.browse(int(product_id)).exists().filtered_domain(self.website_id.sale_product_domain())`

@tivisse

Reply:
You wouldn't be able to do a filtered_domain with a `user_has_group`.
So `filtered` or this way is basically the same.

Why this concern?

Reply:
because if sale_product_domain() is customized, _is_add_to_cart_allowed() should also customized.